openassessment/xblock/grade_mixin.py

```diff
@@ -584,3 +590,101 @@ def _assessment_grade_context(self, assessment):
+        is_peer_assessment = "peer-assessment" in self.assessment_steps
+        is_staff_assessment = "staff-assessment" in self.assessment_steps
+
+        if is_staff_assessment:
+            _assessment_type = "staff_defined"
+        else:
+            # Staff assessment is not a requirement but it can be overrided
+
+            # Check for staff override
+            if complete:
+                # If an assessment is done by staff it always have related annotation
+                _assessment_type = (
+                    "staff_defined"
+                    if "staff_defined" in [
+                        annotation['annotation_type'] for annotation in score["annotations"]
+                    ]
+                    else None
+                )
+
+            # if it is not staff assessment and it also isn't overrided by staff
+            if _assessment_type is None:
+                try:
+                    if not is_peer_assessment:
+                        raise ValueError()
+                    peer_index = workflow["assessment_score_priority"].index("peer")
+                except ValueError:
+                    peer_index = float("inf")
+
+                try:
+                    if not is_self_assessment:
+                        raise ValueError()
+                    self_index = workflow["assessment_score_priority"].index("self")
+                except ValueError:
+                    self_index = float("inf")
+
+                if min(self_index, peer_index) == peer_index:
+                    _assessment_type = "peer_defined"
+                else:
+                    _assessment_type = "self_defined"
+
+        return _assessment_type
+
+    def _get_score_explanation(self, workflow):
+        """
+        Return a string which explains how grade is calculated for an ORA assessment
+        (which is complete i.e all assessments have been done) based on assessment_steps.
+        Args:
+            score (dict): Dict containing points_possible, points_earned and its related attributes
+        Returns:
+            str
+        """
+        score = workflow['score']
+        complete = score is not None
+
+        assessment_type = self._get_assessment_type(workflow)
+        is_self_assessment = "self-assessment" in self.assessment_steps
+        is_peer_assessment = "peer-assessment" in self.assessment_steps
+        is_staff_assessment = "staff-assessment" in self.assessment_steps
+        is_peer_only_assessment = is_peer_assessment and not (is_self_assessment or is_staff_assessment)
+
+        sentences = {
+            "staff": _("The grade for this problem is determined by your Staff Grade."),
+            "peer": _(
+                "The grade for this problem is determined by the median score of "
+                "your Peer Assessments."
+            ),
+            "self": _("The grade for this problem is determined by your Self Assessment.")
+        }
```

Comment:
If you see current version, you would understand why this should remain there. I think this would also be beneficial in future as the grade explanation function become more complex and then it would be very convenient to refer to this dict.

Reply:
Makes sense, lets keep it as is then